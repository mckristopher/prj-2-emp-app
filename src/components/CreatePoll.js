import { connect } from "react-redux";
import { useState } from "react";
import { handleCreatePoll } from '../actions/common';
import { withRouter } from "../util/helper";
import '../css/create-poll.css';

function CreatePoll({ dispatch, router }) {
    const [optionOne, setOptionOne] = useState('');
    const [optionTwo, setOptionTwo] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();

        dispatch(handleCreatePoll([ optionOne, optionTwo ]));

        setOptionOne('');
        setOptionTwo('');

        router.navigate('/home');
    }

    return (
        <div className="create-container">
            <img src="/images/vote.png" className="vote" alt="vote" />
            <h2 className="heading">Would You Rather?..</h2>
            <form onSubmit={handleSubmit}>
                <input placeholder="Option One" type="text" onChange={(e) => setOptionOne(e.target.value)} />
                <input placeholder="Option Two" type="text" onChange={(e) => setOptionTwo(e.target.value)} />
                <button type="submit">SUBMIT</button>
            </form>
        </div>
    )
}

export default withRouter(connect()(CreatePoll));