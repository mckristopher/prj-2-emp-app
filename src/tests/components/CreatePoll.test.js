import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom'


import CreatePoll from '../../components/CreatePoll';

import store from './mock-store';

const navigate = jest.fn()

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

describe('CreatePoll', () => {

    it('should load CreatePoll', () => {
        let component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <CreatePoll />
                </BrowserRouter>
            </Provider>
        );
        expect(component).toMatchSnapshot();
    })

    it('should create new question on submit', () => {
        let component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <CreatePoll />
                </BrowserRouter>
            </Provider>
        );

        let optionOne = component.getByTestId('optionOne'),
        optionTwo = component.getByTestId('optionTwo'),
        submitQuestion = component.getByTestId('submitQuestion');


        fireEvent.change(optionOne, { target: { value: 'React'}});
        fireEvent.change(optionTwo, { target: { value: 'Python'}});
        fireEvent.click(submitQuestion);

        expect(mockedUsedNavigate).toHaveBeenCalledWith('/home');
    })

    it('should enable submit when options are filled', () => {
        let component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <CreatePoll />
                </BrowserRouter>
            </Provider>
        );

        let optionOne = component.getByTestId('optionOne'),
        optionTwo = component.getByTestId('optionTwo'),
        submitQuestion = component.getByTestId('submitQuestion');

        expect(submitQuestion).toBeDisabled();

        fireEvent.change(optionOne, { target: { value: 'React'}});
        fireEvent.change(optionTwo, { target: { value: 'Python'}});

        expect(submitQuestion).not.toBeDisabled();
    })
})