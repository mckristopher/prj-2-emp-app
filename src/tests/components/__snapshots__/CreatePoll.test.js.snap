// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePoll should load CreatePoll 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="create-container"
      >
        <img
          alt="vote"
          class="vote"
          src="/images/vote.png"
        />
        <h2
          class="heading"
        >
          Would You Rather?..
        </h2>
        <form>
          <input
            data-testid="optionOne"
            placeholder="Option One"
            type="text"
          />
          <input
            data-testid="optionTwo"
            placeholder="Option Two"
            type="text"
          />
          <button
            data-testid="submitQuestion"
            disabled=""
            type="submit"
          >
            SUBMIT
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="create-container"
    >
      <img
        alt="vote"
        class="vote"
        src="/images/vote.png"
      />
      <h2
        class="heading"
      >
        Would You Rather?..
      </h2>
      <form>
        <input
          data-testid="optionOne"
          placeholder="Option One"
          type="text"
        />
        <input
          data-testid="optionTwo"
          placeholder="Option Two"
          type="text"
        />
        <button
          data-testid="submitQuestion"
          disabled=""
          type="submit"
        >
          SUBMIT
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
